### PR TITLE
Add generic AbilityOf lookup and friendly ability errors

### PR DESCRIPTION
**Pull request by nchursin**
## Summary
- add generic core.AbilityOf to return typed abilities
- align ability error messages for AbilityTo/AbilityOf with friendly phrasing
- cover behavior with tests including nil actor and multiple abilities

## Testing
- go test ./...

### CODE REVIEW

File: serenity/core/ability.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		return "<nil>"
+	}
+	if t.Kind() == reflect.Pointer {
+		t = t.Elem()
+	}
+	return t.String()
+}
+
+// AbilityNameOf returns the readable name of ability type T without pointer prefix.
+func AbilityNameOf[T abilities.Ability]() string {
```
Comment (nchursin) [inline serenity/core/ability.go]: I think all usage of these can be replaced with AbilityName
